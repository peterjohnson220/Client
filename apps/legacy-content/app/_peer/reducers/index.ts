import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromUpsertDataCutPageReducer from './upsert-data-cut-page.reducer';
import * as fromDataCutValidationReducer from './data-cut-validation.reducer';
import * as fromAssociateCompanyJobReducer from './associate-company-job.reducer';
import * as fromPaymarketExchangeScopeReducer from './paymarket-exchange-scope.reducer';
import * as fromRequestPeerAccessReducer from './request-peer-access.reducer';

// Feature area state
export interface UpsertPeerDataState {
  upsertDataCutPage: fromUpsertDataCutPageReducer.State;
  dataCutValidation: fromDataCutValidationReducer.State;
  associateCompanyJob: fromAssociateCompanyJobReducer.State;
  paymarketExchangeScope: fromPaymarketExchangeScopeReducer.State;
  requestPeerAccess: fromRequestPeerAccessReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  legacy_upsertPeerData: UpsertPeerDataState;
}

// Feature area reducers
export const reducers = {
  upsertDataCutPage: fromUpsertDataCutPageReducer.reducer,
  dataCutValidation: fromDataCutValidationReducer.reducer,
  associateCompanyJob: fromAssociateCompanyJobReducer.reducer,
  paymarketExchangeScope: fromPaymarketExchangeScopeReducer.reducer,
  requestPeerAccess: fromRequestPeerAccessReducer.reducer
};

// Select Feature Area
export const selectUpsertPeerDataState =
  createFeatureSelector<UpsertPeerDataState>('legacy_upsertPeerData');

// Feature Selectors
export const selectUpsertDataCutState =
  createSelector(selectUpsertPeerDataState, (state: UpsertPeerDataState) => state.upsertDataCutPage);
export const selectDataCutValidationState =
  createSelector(selectUpsertPeerDataState, (state: UpsertPeerDataState) => state.dataCutValidation);
export const selectAssociateCompanyJobState =
  createSelector(selectUpsertPeerDataState, (state: UpsertPeerDataState) => state.associateCompanyJob);
export const selectPaymarketExchangeState =
  createSelector(selectUpsertPeerDataState, (state: UpsertPeerDataState) => state.paymarketExchangeScope);
export const selectRequestPeerAccessState =
  createSelector(selectUpsertPeerDataState, (state: UpsertPeerDataState) => state.requestPeerAccess);

// Add Data Cut Selectors
export const getUpsertDataCutAddingDataCut = createSelector(
  selectUpsertDataCutState,
  fromUpsertDataCutPageReducer.getUpsertingDataCut
);
export const getUpsertDataCutAddingDataCutError = createSelector(
  selectUpsertDataCutState,
  fromUpsertDataCutPageReducer.getUpsertingDataCutError
);
export const getUpsertDataCutPageInViewInIframe = createSelector(
  selectUpsertDataCutState,
  fromUpsertDataCutPageReducer.getPageInViewInIframe
);
export const getUpsertDataCutLoadingDataCutDetails = createSelector(
  selectUpsertDataCutState,
  fromUpsertDataCutPageReducer.getLoadingDataCutDetails
);
export const getUpsertDataCutLoadingDataCutError = createSelector(
  selectUpsertDataCutState,
  fromUpsertDataCutPageReducer.getLoadingDataCutDetailsError
);

// Data Cut Validation Selectors
export const {
  selectAll: getDataCutValidationInfo
} = fromDataCutValidationReducer.adapter.getSelectors(selectDataCutValidationState);
export const getDataCutValidationInfoLoading =
  createSelector(selectDataCutValidationState, fromDataCutValidationReducer.getLoading);
export const getDataCutValidationInfoLoadingError
  = createSelector(selectDataCutValidationState, fromDataCutValidationReducer.getLoadingError);
export const getEmployeeSimilarityError
  = createSelector(selectDataCutValidationState, fromDataCutValidationReducer.getEmployeeSimilarityError);
export const getEmployeeCheckPassed = createSelector(selectDataCutValidationState,
  fromDataCutValidationReducer.getEmployeeSimilarityPassed);
export const getIsEmployeeSimilarityLoading = createSelector(selectDataCutValidationState,
  fromDataCutValidationReducer.getValidatingEmployeeSimilarity);
// Associate Company Job Selectors
export const getExchangeSearchResult = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getExchangeJobs
);

export const getIsLoading = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getIsLoading
);

export const getHasLoadingError = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getHasLoadingError
);

export const getIsAdding = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getIsAdding
);

export const getHasAddingError = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getHasAddingError
);

export const getCompanyJob = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getCompanyJob
);

export const getLoadingExchangeDictionary = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getLoadingExchangeDictionary
);

export const getLoadingExchangeDictionaryError = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getLoadingExchangeDictionaryError
);

export const getExchangeDictionaryForCompany = createSelector(
  selectAssociateCompanyJobState,
  fromAssociateCompanyJobReducer.getExchangeDictionary
);

// Paymarket Exchange Scope Selectors
export const getExchangeScopesIsLoading = createSelector(
  selectPaymarketExchangeState,
  fromPaymarketExchangeScopeReducer.getExchangeScopesIsLoading
);

export const getExchangeScopeLoadingError = createSelector(
  selectPaymarketExchangeState,
  fromPaymarketExchangeScopeReducer.getExchangeScopeLoadingError
);

export const getExchangeScopes = createSelector(
  selectPaymarketExchangeState,
  fromPaymarketExchangeScopeReducer.getExchangeScopes
);

export const getSelectedExchangeScopes = createSelector(
  selectPaymarketExchangeState,
  fromPaymarketExchangeScopeReducer.getSelectedExchangeScopes
);

// Request Peer Access Selectors
export const getRequestingPeerAccess = createSelector(
  selectRequestPeerAccessState,
  fromRequestPeerAccessReducer.getRequestingPeerAccess
);

export const getRequestingPeerAccessError = createSelector(
  selectRequestPeerAccessState,
  fromRequestPeerAccessReducer.getRequestingPeerAccessError
);

