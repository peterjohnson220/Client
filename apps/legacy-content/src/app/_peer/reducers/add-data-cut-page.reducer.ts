import * as fromAddDataCutPageActions from '../actions/add-data-cut-page.actions';

import { ExchangeDataCutBaseFilter } from 'libs/models';

export interface State {
  addingDataCut: boolean;
  addingDataCutError: boolean;
  baseExchangeDataCutFilter: ExchangeDataCutBaseFilter;
}

// Initial State
export const initialState: State = {
  addingDataCut: false,
  addingDataCutError: false,
  baseExchangeDataCutFilter: null
};

// Reducer
export function reducer(
  state = initialState,
  action: fromAddDataCutPageActions.Actions
): State {
  switch (action.type) {
    case fromAddDataCutPageActions.ADDING_DATA_CUT: {
      return {
        ...state,
        addingDataCut: true,
        addingDataCutError: false
      };
    }
    case fromAddDataCutPageActions.ADDING_DATA_CUT_SUCCESS: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: false
      };
    }
    case fromAddDataCutPageActions.ADDING_DATA_CUT_ERROR: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: true
      };
    }
    case fromAddDataCutPageActions.CANCEL_ADD_DATA_CUT: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: false
      };
    }
    case fromAddDataCutPageActions.LOADING_BASE_FILTER_SUCCESS: {
      return {
        ...state,
        baseExchangeDataCutFilter: action.payload
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getAddingDataCut = (state: State) => state.addingDataCut;
export const getAddingDataCutError = (state: State) => state.addingDataCutError;
export const getBaseExchangeDataCutFilter = (state: State) => state.baseExchangeDataCutFilter;
