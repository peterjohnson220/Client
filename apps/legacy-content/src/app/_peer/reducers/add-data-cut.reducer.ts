import * as fromAddDataCutActions from '../actions/add-data-cut.actions';

export interface State {
  addingDataCut: boolean;
  addingDataCutError: boolean;
}

// Initial State
export const initialState: State = {
  addingDataCut: false,
  addingDataCutError: false
};

// Reducer
export function reducer(
  state = initialState,
  action: fromAddDataCutActions.Actions
): State {
  switch (action.type) {
    case fromAddDataCutActions.ADDING_DATA_CUT: {
      return {
        ...state,
        addingDataCut: true,
        addingDataCutError: false
      };
    }
    case fromAddDataCutActions.ADDING_DATA_CUT_SUCCESS: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: false
      };
    }
    case fromAddDataCutActions.ADDING_DATA_CUT_ERROR: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: true
      };
    }
    case fromAddDataCutActions.CANCEL_ADD_DATA_CUT: {
      return {
        ...state,
        addingDataCut: false,
        addingDataCutError: false
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getAddingDataCut = (state: State) => state.addingDataCut;
export const getAddingDataCutError = (state: State) => state.addingDataCutError;
