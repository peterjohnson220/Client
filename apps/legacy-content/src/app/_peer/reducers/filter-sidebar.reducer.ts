import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';

import { FilterAggregateGroup } from 'libs/models/peer/aggregate-filters';
import { ExchangeDataCutFilter, FilterAggregateSelection } from 'libs/models/peer';

import * as fromFilterSidebarActions from '../actions/filter-sidebar.actions';

// Extended entity state
export interface State extends EntityState<FilterAggregateGroup> {
  loading: boolean;
  loadingError: boolean;
  limitToPayMarket: boolean;
  payMarket: any;
  selections: FilterAggregateSelection;
}

// Create entity adapter
export const adapter: EntityAdapter<FilterAggregateGroup> = createEntityAdapter<FilterAggregateGroup>({
  selectId: (filter: FilterAggregateGroup) => filter.MetaData.PeerFilter
});

// Initial State
export const initialState: State = adapter.getInitialState({
  loading: false,
  loadingError: false,
  limitToPayMarket: true,
  payMarket: null,
  selections: {
    Exchanges: [],
    States: [],
    Cities: [],
    Companies: [],
    CompanyIndustries: [],
    ExchangeJobFamilies: [],
    ExchangeJobLevels: [],
    ExchangeJobFLSAStatuses: []
  }
});

// Reducer
export function reducer(state = initialState, action: fromFilterSidebarActions.Actions): State {
  switch (action.type) {
    case fromFilterSidebarActions.LOADING_PEER_FILTERS: {
      return {
        ...adapter.removeAll(state),
        loading: true
      };
    }
    case fromFilterSidebarActions.LOADING_PEER_FILTERS_SUCCESS: {
      const filters: FilterAggregateGroup[] = action.payload.response;
      const filter: ExchangeDataCutFilter = action.payload.filter;
      const filtersWithSelections: FilterAggregateGroup[] = filters.map(f => {
        return {
          ...f,
          Selections: filter[ f.MetaData.FilterProp ]
        };
      });
      return {
        ...adapter.addAll(filtersWithSelections, state),
        loading: false
      };
    }
    case fromFilterSidebarActions.LOADING_PEER_FILTERS_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    case fromFilterSidebarActions.TOGGLE_AGGREGATE_SELECTED: {
      const selectionsCopy = { ...state.selections };
      const selectionInfo = action.payload;

      let selectionsGroupCopy = [...selectionsCopy[selectionInfo.AggregateGroup]];

      if (selectionsGroupCopy.some(s => s === selectionInfo.AggregateItem)) {
        selectionsGroupCopy = selectionsGroupCopy.filter(s => s !== selectionInfo.AggregateItem);
      } else {
        selectionsGroupCopy.push(selectionInfo.AggregateItem);
      }

      selectionsCopy[selectionInfo.AggregateGroup] = selectionsGroupCopy;

      return {
        ...state,
        selections: selectionsCopy
      };
    }
    case fromFilterSidebarActions.TOGGLE_LIMIT_TO_PAYMARKET: {
      return {
        ...state,
        limitToPayMarket: !state.limitToPayMarket
      };
    }
    case fromFilterSidebarActions.LOAD_PAYMARKET_INFORMATION_SUCCESS: {
      return {
        ...state,
        payMarket: action.payload
      };
    }
    case fromFilterSidebarActions.CLEAR_SELECTIONS: {
      return {
        ...state,
        selections: initialState.selections
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
export const getSelections = (state: State) => state.selections;
export const getLimitToPayMarket = (state: State) => state.limitToPayMarket;
export const getPayMarket = (state: State) => state.payMarket;
