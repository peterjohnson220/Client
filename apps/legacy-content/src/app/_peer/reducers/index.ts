import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromPeerMapReducer from './peer-map.reducer';
import * as fromPeerFiltersReducer from './peer-filters.reducer';

// Feature area state
export interface PeerDataState {
  map: fromPeerMapReducer.State;
  filters: fromPeerFiltersReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  peerData: PeerDataState;
}

// Feature area reducers
export const reducers = {
  map: fromPeerMapReducer.reducer,
  filters: fromPeerFiltersReducer.reducer
};

// Select Feature Area
export const selectPeerDataState = createFeatureSelector<PeerDataState>('peerData');

// Feature Selectors
export const selectMapState = createSelector(selectPeerDataState, (state: PeerDataState) => state.map);
export const selectPeerFiltersState = createSelector(selectPeerDataState, (state: PeerDataState) => state.filters);

// Map Data Selectors
export const getPeerMapLoading = createSelector(selectMapState, fromPeerMapReducer.getLoading);
export const getPeerMapLoadingError = createSelector(selectMapState, fromPeerMapReducer.getLoadingError);
export const getPeerMapSummary = createSelector(selectMapState, fromPeerMapReducer.getMapSummary);
export const getPeerMapFilter = createSelector(selectMapState, fromPeerMapReducer.getMapFilter);
export const getPeerMapCollection = createSelector(selectMapState, fromPeerMapReducer.getMapCollection);
export const getPeerMapBounds = createSelector(selectMapState, fromPeerMapReducer.getMapBounds);
export const canLoadPeerMap = createSelector(selectMapState, fromPeerMapReducer.canLoadMap);
export const peerMapShowNoData = createSelector(selectMapState, fromPeerMapReducer.showNoData);


// Filters
export const {
  selectAll: getPeerFilters
} = fromPeerFiltersReducer.adapter.getSelectors(selectPeerFiltersState);

export const getPeerFiltersLoading = createSelector(selectPeerFiltersState, fromPeerFiltersReducer.getLoading);
export const getPeerFiltersLoadingError = createSelector(selectPeerFiltersState, fromPeerFiltersReducer.getLoadingError);
