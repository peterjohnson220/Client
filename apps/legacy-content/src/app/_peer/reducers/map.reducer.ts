import { FeatureCollection, Point } from 'geojson';

import { ExchangeMapSummary } from 'libs/models/peer';

import * as fromPeerMapActions from '../actions/map.actions';

export interface State {
  mapCollection: FeatureCollection<Point>;
  mapSummary: ExchangeMapSummary;
  mapFilter: any;
  mapBounds: number[];
  loading: boolean;
  loadingError: boolean;
  shouldUpdateBounds: boolean;
  isInitialLoad: boolean;
  initialMapMoveComplete: boolean;
  maxZoom: number;
}

// Initial State
export const initialState: State = {
  mapCollection: null,
  mapFilter: {
    TopLeft: null,
    BottomRight: null,
    ClusterPrecision: 8
  },
  mapSummary: null,
  mapBounds: [0, 0, 0, 0],
  loading: false,
  loadingError: false,
  shouldUpdateBounds: true,
  isInitialLoad: true,
  initialMapMoveComplete: false,
  maxZoom: 7
};

// Reducer
export function reducer(state = initialState, action: fromPeerMapActions.Actions): State {
  switch (action.type) {
    case fromPeerMapActions.LOADING_PEER_MAP: {
      return {
        ...state,
        mapCollection: null,
        loading: true,
        loadingError: false
      };
    }
    case fromPeerMapActions.LOADING_PEER_MAP_SUCCESS: {
      const mapSummary: ExchangeMapSummary = action.payload.MapSummary;
      const mapCollection: FeatureCollection<Point> = {
        type: 'FeatureCollection',
        features: action.payload.FeatureCollection
      };
      const mapFilter = {
        ...state.mapFilter
      };
      const newState = {
        ...state,
        mapCollection: mapCollection,
        mapSummary: mapSummary,
        loading: false,
        loadingError: false,
        mapFilter: mapFilter,
        isInitialLoad: false
      };
      if (state.isInitialLoad) {
        const tl = mapSummary.TopLeft;
        const br = mapSummary.BottomRight;
        newState.mapBounds = [tl.Lon, br.Lat, br.Lon, tl.Lat];
        newState.mapFilter.TopLeft = tl;
        newState.mapFilter.BottomRight = br;
      }
      return newState;
    }
    case fromPeerMapActions.LOADING_PEER_MAP_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    case fromPeerMapActions.INITIAL_MAP_MOVE_COMPLETE: {
      return {
        ...state,
        mapFilter: updateMapFilterBounds(state, action.payload.bounds),
        initialMapMoveComplete: true,
        maxZoom: 17
      };
    }
    case fromPeerMapActions.UPDATE_PEER_MAP_FILTER_BOUNDS: {
      return {
        ...state,
        mapFilter: updateMapFilterBounds(state, action.payload.bounds)
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getMapSummary = (state: State) => state.mapSummary;
export const getMapFilter = (state: State) => state.mapFilter;
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
export const getMapCollection = (state: State) => state.mapCollection;
export const getMapBounds = (state: State) => state.mapBounds;
export const getMaxZoom = (state: State) => state.maxZoom;
export const getInitialMapMoveComplete = (state: State) => state.initialMapMoveComplete;
export const canLoadMap = (state: State) => !state.isInitialLoad && !state.loading;
export const showNoData = (state: State) => !state.loading && !state.isInitialLoad && state.mapCollection.features.length === 0;

function swapBounds(bounds: any): any {
  const ne = bounds._ne;
  const sw = bounds._sw;
  const swappedBounds = {
    TopLeft: {
      Lat: enforceBoundsLimit(ne.lat),
      Lon: enforceBoundsLimit(sw.lng)
    },
    BottomRight: {
      Lat: enforceBoundsLimit(sw.lat),
      Lon: enforceBoundsLimit(ne.lng)
    }
  };
  return swappedBounds;
}

function enforceBoundsLimit(coordinate: number) {
  return coordinate > 180 ? 180 : coordinate < -180 ? -180 : coordinate;
}

function updateMapFilterBounds(state, bounds) {
  const swappedBounds = swapBounds(bounds);

  return {
    ...state.mapFilter,
    TopLeft: swappedBounds.TopLeft,
    BottomRight: swappedBounds.BottomRight
  };
}
