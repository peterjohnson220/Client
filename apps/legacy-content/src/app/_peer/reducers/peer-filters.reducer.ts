import { createEntityAdapter, EntityAdapter, EntityState } from '@ngrx/entity';

import * as fromPeerFilterActions from '../actions/peer-filters.actions';
import { FilterAggregateGroup } from 'libs/models/peer/aggregate-filters';

// Extended entity state
export interface State extends EntityState<FilterAggregateGroup> {
  loading: boolean;
  loadingError: boolean;
}

// Create entity adapter
export const adapter: EntityAdapter<FilterAggregateGroup> = createEntityAdapter<FilterAggregateGroup>({
  selectId: (filter: FilterAggregateGroup) => filter.MetaData.FilterType
});

// Initial State
export const initialState: State = adapter.getInitialState({
  loading: false,
  loadingError: false
});

// Reducer
export function reducer(state = initialState, action: fromPeerFilterActions.Actions): State {
  switch (action.type) {
    case fromPeerFilterActions.LOADING_PEER_FILTERS: {
      return {
        ...adapter.removeAll(state),
        loading: true
      };
    }
    case fromPeerFilterActions.LOADING_PEER_FILTERS_SUCCESS: {
      const filters: FilterAggregateGroup[] = action.payload;
      return {
        ...adapter.addAll(filters, state),
        loading: false
      };
    }
    case fromPeerFilterActions.LOADING_PEER_FILTERS_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
