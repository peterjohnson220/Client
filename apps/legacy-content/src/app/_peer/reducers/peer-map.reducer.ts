import { FeatureCollection, Point } from 'geojson';
import { ExchangeMapFilter, ExchangeMapSummary } from 'libs/models/peer';

import * as fromPeerMapActions from '../actions/peer-map.actions';

export interface MapBounds {
  center: number[];
  bounds: number[];
}

export interface State {
  mapCollection: FeatureCollection<Point>;
  mapSummary: ExchangeMapSummary;
  mapFilter: ExchangeMapFilter;
  mapBounds: MapBounds;
  loading: boolean;
  loadingError: boolean;
  boundsChanged: boolean;
}

// Initial State
export const initialState: State = {
  mapCollection: null,
  mapFilter: {
    States: [],
    Cities: [],
    ClusterPrecision: 1
  },
  mapSummary: null,
  mapBounds: {
    center: [],
    bounds: []
  },
  loading: false,
  loadingError: false,
  boundsChanged: false
}

// Reducer
export function reducer(state = initialState, action: fromPeerMapActions.Actions): State {
  switch (action.type) {
    case fromPeerMapActions.LOADING_PEER_MAP: {
      return {
        ...state,
        mapCollection: null,
        mapSummary: null,
        loading: true,
        loadingError: false,
        boundsChanged: false
      };
    }
    case fromPeerMapActions.LOADING_PEER_MAP_SUCCESS: {
      const mapSummary: ExchangeMapSummary = action.payload.MapSummary;
      const mapCollection: FeatureCollection<Point> = {
        type: 'FeatureCollection',
        features: action.payload.FeatureCollection
      };
      const tl = mapSummary.TopLeft;
      const br = mapSummary.BottomRight;
      const mapFilter = {
        ...state.mapFilter,
        TopLeft: tl,
        BottomRight: br
      };
      return {
        ...state,
        mapCollection: mapCollection,
        mapSummary: mapSummary,
        loading: false,
        loadingError: false,
        mapFilter: mapFilter,
        mapBounds: {
          center: [mapSummary.Center.Lon, mapSummary.Center.Lat],
          bounds: [tl.Lon, br.Lat, br.Lon, tl.Lat]
        },
        boundsChanged: true
      };
    }
    case fromPeerMapActions.LOADING_PEER_MAP_ERROR: {
      return {
        ...state,
        loading: false,
        loadingError: true,
        boundsChanged: false
      };
    }
    case fromPeerMapActions.LOADING_INITIAL_PEER_MAP_FILTER_SUCCESS: {
      const mapFilterFromServer: ExchangeMapFilter = action.payload;
      const mapFilter = {
        ...mapFilterFromServer,
        ClusterPrecision: initialState.mapFilter.ClusterPrecision
      };
      return {
        ...state,
        mapFilter: mapFilter,
        boundsChanged: false
      };
    }
    case fromPeerMapActions.UPDATE_PEER_MAP_FILTER_BOUNDS: {
      const bounds = swapBounds(action.payload.bounds);
      const zoom = action.payload.zoom;
      const zoomPercentage = zoom / 20;
      const prec = zoom <= 0 ? 1 : Math.round(zoomPercentage * 12);
      const mapFilter = {
        ...state.mapFilter,
        TopLeft: bounds.TopLeft,
        BottomRight: bounds.BottomRight,
        ClusterPrecision: prec
      };
      return {
        ...state,
        mapFilter: mapFilter,
        boundsChanged: false
      };
    }
    case fromPeerMapActions.SKIP_MAP_QUERY: {
      return {
        ...state,
        boundsChanged: false
      };
    }
    default: {
      return state;
    }
  }
}

// Selector Functions
export const getMapSummary = (state: State) => state.mapSummary;
export const getMapFilter = (state: State) => state.mapFilter;
export const getLoading = (state: State) => state.loading;
export const getLoadingError = (state: State) => state.loadingError;
export const getMapCollection = (state: State) => state.mapCollection;
export const getMapBounds = (state: State) => state.mapBounds.bounds;
export const getMapCenter = (state: State) => state.mapBounds.center;
export const getBoundsChanged = (state: State) => state.boundsChanged;


function swapBounds(bounds: any): any {
  const ne = bounds._ne;
  const sw = bounds._sw;
  const swappedBounds = {
    TopLeft: {
      Lat: ne.lat,
      Lon: sw.lng
    },
    BottomRight: {
      Lat: sw.lat,
      Lon: ne.lng
    }
  };
  return swappedBounds;
}
