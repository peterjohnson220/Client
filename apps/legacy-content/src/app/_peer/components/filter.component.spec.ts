import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';

import { combineReducers, Store, StoreModule } from '@ngrx/store';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import {
  generateMockFilterAggregateGroup, generateMockFilterAggregateItem, generateMockFilterAggregateMetaData,
  generateMockUpdateFilterSelectionsModel
} from 'libs/models/peer/aggregate-filters';

import * as fromPeerDataReducer from '../reducers';
import { FilterComponent } from './filter.component';

describe('Legacy Content - Peer - Filter Component', () => {
  let fixture: ComponentFixture<FilterComponent>;
  let instance: FilterComponent;
  let store: Store<fromRootState.State>;
  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peerData: combineReducers(fromPeerDataReducer.reducers)
        })
      ],
      declarations: [
        FilterComponent
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);

    spyOn(store, 'dispatch');

    fixture = TestBed.createComponent(FilterComponent);
    instance = fixture.componentInstance;
  });

  it('should set selections from filter on init', () => {
    const mockFilterAggregateGroup = generateMockFilterAggregateGroup();
    mockFilterAggregateGroup.Aggregates.push(generateMockFilterAggregateItem('MockSelection'));
    instance.filter = mockFilterAggregateGroup;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should emit the selectionChanged event when the handleValueChangeEvent event is triggered', () => {
    const mockMetaData = {...generateMockFilterAggregateMetaData(), FilterProp: 'Exchanges'};
    const mockFilterAggregateGroup = {...generateMockFilterAggregateGroup(), MetaData: mockMetaData};
    const mockUpdateFilterSelection = generateMockUpdateFilterSelectionsModel();
    instance.filter = mockFilterAggregateGroup;

    spyOn(instance.selectionChanged, 'emit');

    fixture.detectChanges();

    instance.handleValueChangeEvent(mockUpdateFilterSelection);

    expect(instance.selectionChanged.emit).toBeCalledWith(mockUpdateFilterSelection);
  });
});
