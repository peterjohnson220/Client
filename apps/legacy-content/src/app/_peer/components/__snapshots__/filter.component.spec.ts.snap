// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Legacy Content - Peer - Filter Component should set selections from filter on init 1`] = `
<pf-peer-data-cut-filter
  data={[Function Array]}
  filter={[Function Object]}
  list={[Function ElementRef]}
  selectionChanged={[Function EventEmitter]}
  selections={[Function Array]}
  source={[Function Array]}
>
  <label
    for="MockId"
  >
    MockLabel
  </label>
  <kendo-multiselect
    id="MockId"
    textfield="Item"
    valuefield="Item"
  >
    
  
  </kendo-multiselect>
  
</pf-peer-data-cut-filter>
`;
